app/controllers/users_controller.rb

```diff
@@ -311,7 +311,7 @@ def create
       return fail_with("login.reserved_username")
     end
 
-    if user = User.where(staged: true).find_by(email: params[:email].strip.downcase)
+    if user = User.where(staged: true).find_by_email(params[:email].strip.downcase)
```

Comment:
> find_by_email is not a scope so it will not filter on staged: true

I'm a bit confused, because it does seem to filter. I've added a test in my latest commit which passes with the line as it is, but fails if it's changed to `User.find_by_email(params[:email].strip.downcase)`, meaning the `.where(staged: true)` is doing something.

I can also see it filtering by staged: true in the console:

```
pry(main)> User.where(staged: true).find_by_email('no_email')
  UserEmail Load (0.3ms)  SELECT  "user_emails".* FROM "user_emails" WHERE "user_emails"."email" = 'no_email' LIMIT 1  [["email", "no_email"]]
  User Load (0.3ms)  SELECT  "users".* FROM "users" WHERE "users"."staged" = 't' AND "users"."id" = -1 LIMIT 1  [["staged", true], ["id", -1]]
=> nil
```